tests/queries/0_stateless/02267_file_globs_schema_inference.sql

```diff
@@ -5,7 +5,7 @@ insert into function file('02267_data3.jsonl') select * from numbers(0);
 insert into function file('02267_data4.jsonl') select 1 as x;
 select * from file('02267_data*.jsonl') order by x;
 
-insert into function file('02267_data1.jsonl', 'TSV') select 1 as x;
+insert into function file('02267_data4.jsonl', 'TSV') select 1 as x;
 insert into function file('02267_data1.jsonl', 'TSV') select [1,2,3] as x;
 
 select * from file('02267_data*.jsonl') settings schema_inference_use_cache_for_file=0; --{serverError INCORRECT_DATA}
```

Comment:
```suggestion
select * from file('02267_data*.jsonl') settings schema_inference_use_cache_for_file=0; -- {serverError INCORRECT_DATA}

-- cleanup
insert into function file('02267_data1.jsonl', 'TSV') select * from numbers(0) SETTINGS engine_file_truncate_on_insert = 1;
insert into function file('02267_data2.jsonl', 'TSV') select * from numbers(0) SETTINGS engine_file_truncate_on_insert = 1;
insert into function file('02267_data3.jsonl', 'TSV') select * from numbers(0) SETTINGS engine_file_truncate_on_insert = 1;
insert into function file('02267_data4.jsonl', 'TSV') select * from numbers(0) SETTINGS engine_file_truncate_on_insert = 1;


```

Reply:
Good idea, but I would prefer using `schema_inference_use_cache_for_file = 0` on the first insertion for each file rather than adding this cleanup. I will add it